### ISSUE TITLE
Updates to heat in built environment has caused PVT to break

### ISSUE THREAD

**Issue by mabijkerk** [quintel/etengine#1400]
Setting `households_heater_heatpump_pvt_electricity_share` to 100.0 on `production` increases final demand for solar thermal PVT, which determines the demand of the `households_solar_pvt_solar_radiation` node:

![Screenshot 2024-02-07 at 10 06 45](https://github.com/quintel/etengine/assets/67338510/a034c8e6-19c8-4bcb-a7c4-5cd2d6375283)

On `master`, the same demand is passed to the outgoing `solar_thermal` edge of the `households_solar_pvt_solar_radiation` node, but the demand on the node itself is not picked up:

![Screenshot 2024-02-07 at 10 07 52](https://github.com/quintel/etengine/assets/67338510/c00677d5-cee7-4321-a3dc-1a7b59a042fe)

This issue has arised since the latest changes to heat in the built environment. No changes have been made to the graph structure, though the inputs have been changed. From: ...

```
    EACH(
      UPDATE(EDGE(households_space_heater_heatpump_pvt_electricity_aggregator,households_useful_demand_for_space_heating_after_insulation), share, DIVIDE(USER_INPUT(),100)),
      UPDATE(EDGE(households_water_heater_heatpump_pvt_electricity_aggregator,households_useful_demand_for_hot_water_after_solar_heater), share, DIVIDE(USER_INPUT(),100)),
      )
```
... to ...
```
    EACH(
      UPDATE(V(households_space_heater_heatpump_pvt_electricity, fever), share_in_group, DIVIDE(USER_INPUT(),100.0)),
      UPDATE(V(households_water_heater_heatpump_pvt_electricity, fever), share_in_group, DIVIDE(USER_INPUT(),100.0))
    )
```

@noracato is this something you (and I) can look into?

**Comment by noracato** [comment]
> No changes have been made to the graph structure

Correct me if I'm wrong, but the edge from the solar PVT towards the local available electricity seems to have changed from share to flexible. 

Maybe it has nothing to do with it at all. I can check what is happening inside the calculation to give us more insight!

**Comment by mabijkerk** [comment]
Unfortunately it is not related to the edge type, apparently this was something I had tested locally and not commited, because every else the type is `share` instead of `flexible`. With both types the problem persists.

**Comment by noracato** [comment]
This issue had the same cause as #1390. Instead of adding even more manual install checks for Merit participant, I decided to pull Fever's share/pair calculations (between consumers and producers) forward to an early setup stage.

This means the Causality calculations are now in the following order:
1. User values are set on the graph
2. Causality: early setup for Fever. Fever calculates and sets consumer/producer shares based on `share_in_group` and `number_of_units` as per usual
3. Yearly graph calculations: set all demands and shares that could be calulated - _this means all basic heat demands are now known throughout the graph_
4. Causality: setup all other participants
5. Causality: run calculations per hour
6. Causality: reconciliation
7. Yearly graph calculations: fill in rest of flows

Before 2 was part of 4, which meant not all demands were known and Merit did not know whether some participants should be joining the calculation.

The fix I did initially for #1390 has been reversed.
